### PR TITLE
🐛 Source Marketo: improve field conversion to conform schema

### LINKED ISSUE #7776: Fix BigQuery destination normalization to accept any date-time format.
A user using a Marketo to BigQuery connection is facing a bug during basic normalization (slack message [here](https://airbytehq.slack.com/archives/C01MFR03D5W/p1636366826257700)).
It appears that the Marketo connector is outputting a date-time format which is not properly handled by BigQuery leading to the following normalization error: 
`Invalid timestamp: '2018-09-07T17:37:18Z+0000'`.

This issue is closely related to a similar normalization problem with Snowflake (https://github.com/airbytehq/airbyte/issues/5713)

## Environment
- **Airbyte version**:  0.30.33-alpha
- **OS Version / Instance**: n1-standard-2
- **Deployment**: Docker
- **Source Connector and version**: airbyte/source-marketo:0.1.0
- **Destination Connector and version**: airbyte/destination-bigquery:0.5.0
- **Severity**:  High
- **Step where error happened**: Sync job

## Current Behavior
Marketo source outputs date-time format such as `2018-09-07T17:37:18Z+0000` in the `createdAt` field of the Program's stream records.
Normalization fails with `Invalid timestamp: '2018-09-07T17:37:18Z+0000'` error.

## Expected Behavior
BigQuery normalization should handle casting this format to timestamp.

## Logs
[logs-18-2-marketo.txt](https://github.com/airbytehq/airbyte/files/7503514/logs-18-2-marketo.txt)

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
1. `def normalize_datetime` - I propose make date-time normalization more explicit than just trim chars.

2. `MarketoExportBase.parse_response` - improved to be more robust for cases like in (3).

3 . For new stream `activities_new_lead` we get different field names on `discover` and `read` steps WE NEED TO FIX IT!!!
    Tthis happened because discover and read return different names ( which probaly need to be normalized to the same name)

      'API Method Name' -> 'a_p_i_method_name'
      'api method name' -> 'apimethodname'
      
      'Request Id' -> 'request_id'
      "request id" -> 'requestid'

      'Modifying User' -> 'modifying_user'
      'modifying user' -> 'modifyinguser'
      
      WE NEED TO FIX IT !!!!

4. Current implementation of `utils.clean_string` pretty 'magic' and can give ugly-looking results:

     'Step ID' -> 'step_i_d'
     "Campaign Run ID" -> 'campaign_run_i_d'
     "SFDC Type" -> "s_f_d_c_type"
     "Webpage ID" -> "webpage_i_d"
     
     but we can live with it 

I think we can to get rid of `utils.clean_string` and just use `s.lower().replace(' ', '_')`

```
Orig Value                         utils.clean_string            s.lower().replace(' ', '_')
-----------------------------------------------------------------------------------------------
Add to List                      addto_list                      add_to_list
Add to Nurture                   addto_nurture                   add_to_nurture
Add to Opportunity               addto_opportunity               add_to_opportunity
Change Status in Progression     change_statusin_progression     change_status_in_progression
Merged in Sales                  mergedin_sales                  merged_in_sales
Push Lead to Marketo             push_leadto_marketo             push_lead_to_marketo
Received by                      receivedby                      received_by
Received Forward to Friend Email received_forwardto_friend_email received_forward_to_friend_email
Remove from List                 removefrom_list                 remove_from_list
Remove from Opportunity          removefrom_opportunity          remove_from_opportunity
Reply to Sales Email             replyto_sales_email             reply_to_sales_email
Sent by                          sentby                          sent_by
Sent Forward to Friend Email     sent_forwardto_friend_email     sent_forward_to_friend_email
```

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-marketo/source_marketo/source.py [hunk 1]
```diff
@@ -336,14 +346,10 @@ def get_json_schema(self) -> Mapping[str, Any]:
             "activityDate": {"type": ["null", "string"], "format": "date-time"},
             "activityTypeId": {"type": ["null", "integer"]},
             "campaignId": {"type": ["null", "integer"]},
-            "attributes": {"type": ["null", "string"]},
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-marketo/source_marketo/source.py:339]: Why deleted attributes?

File: airbyte-integrations/connectors/source-marketo/source_marketo/source.py [hunk 2]
```diff
@@ -336,14 +346,10 @@ def get_json_schema(self) -> Mapping[str, Any]:
...
+            "primaryAttributeValueId": {"type": ["null", "string"]},
+            "primaryAttributeValue": {"type": ["null", "string"]},
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-marketo/source_marketo/source.py:350]: Need add `primaryAttributeName`

File: airbyte-integrations/connectors/source-marketo/source_marketo/source.py [hunk 3]
```diff
@@ -435,8 +441,8 @@ def request_params(
 
     def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
         for record in super().parse_response(response, stream_state, **kwargs):
-            # delete +00:00 part from the end of createdAt and updatedAt
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-marketo/source_marketo/source.py:438]: Why deleted comment?

**Review by lazebnyi — APPROVED** [review]
LGTM! Few minor comments.

File: airbyte-integrations/connectors/source-marketo/source_marketo/source.py (hunk 3 above)
Comment (grubberr) [reply airbyte-integrations/connectors/source-marketo/source_marketo/source.py:438]: fixed, reverted back

File: airbyte-integrations/connectors/source-marketo/source_marketo/source.py (hunk 1 above)
Comment (grubberr) [reply airbyte-integrations/connectors/source-marketo/source_marketo/source.py:339]: as I understand we are removing `attributes` from record object

https://github.com/airbytehq/airbyte/blob/65ef9295d16691b0889492c9f6331bc3c1668dbb/airbyte-integrations/connectors/source-marketo/source_marketo/source.py#L223

we need only flattened fields from `attributes`
but `attributes` itself can be removed

maybe I missed something ?

File: airbyte-integrations/connectors/source-marketo/source_marketo/source.py (hunk 2 above)
Comment (grubberr) [reply airbyte-integrations/connectors/source-marketo/source_marketo/source.py:350]: I have dumped the header of all 'activities_*' streams and found that:

1) it's the same for all steams
2) it consists of next fields:

```
marketoGUID,
leadId,
activityDate,
activityTypeId,
campaignId,
primaryAttributeValueId,
primaryAttributeValue,
attributes
```

I as you can see there is no `primaryAttributeName` field 
I have also not found any references of this field in other places like this
https://developers.marketo.com/rest-api/lead-database/activities/

I decided to remove.

What do you think ?

Comment (lazebnyi) [reply airbyte-integrations/connectors/source-marketo/source_marketo/source.py:350]: Ok. We can remove it.

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-marketo/source_marketo/utils.py [hunk 4]
```diff
@@ -37,8 +37,15 @@ def clean_string(string: str) -> str:
     "updatedat" -> "updatedat"
     """
 
-    abbreviations = ("URL", "GUID", "IP")
-    if any(map(lambda w: w in string, abbreviations)):
+    fix = {
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-marketo/source_marketo/utils.py:40]: can we have unit tests?

Comment (grubberr) [reply airbyte-integrations/connectors/source-marketo/source_marketo/utils.py:40]: I have added more cases for this unit_test 'test_clean_string'

**Review by antixar — APPROVED** [review]
LGTM!
Please publish your changes

**Pull request merged**